### ISSUE TITLE
Cannot Start on Ubuntu 14 in Vagrant on Windows 

### ISSUE THREAD

**Issue by OmarShehata** [superscriptjs/superscript#273]
I'm following the instructions on the [quick start](http://superscriptjs.com/starter/quickstart) page, and after creating a bot and running `parse` I get:

```
OpenError: IO error: systemDB/MANIFEST-000001: Invalid Argument 
at /usr/local/lib/node_modules/superscript/node_modules/sfacts/node_modules/level/node_modules/level-packager/node_modules/levelup/lib/levelup.js:114:34
```

Now because I'm running in Vagrant I did run into some issues prior to this. I had to run `npm install` with the `--no-bin-links` because symlinks fail on Vagrant. 

I don't know if it's relevant, but another weird issue when running `npm install` for superscript is that it fails to install **conceptnet**. It seems like it's trying to install it from the npm registry. This is the output:

```
npm http 404 https://registry.npmjs.org/conceptnet
npm ERR! TypeError: Object.keys called on non-object
````

But if I just run `npm install silentrob/conceptnet` that works. Although, now that I check, I don't see it in node_modules/superscript/node_modules. 

Any ideas?

**Edit/Update:** 

Followed the fix here to get Vagrant to use symlinks:

https://github.com/mitchellh/vagrant/issues/713#issuecomment-17296765

Now `sudo npm install superscript -g` runs without an error.  However, it doesn't seem to edit the PATH? (So bot-init says it's not found). I can fix this by adding `/usr/local/lib/node_modules/superscript/bin` to the PATH and then calling `bot-init.js` and `parse.js`

And `sudo npm install` inside the bot is giving me EPERM errors, (even when running as admin and with sudo) I think this should fix it https://harvsworld.com/2015/how-to-fix-npm-install-errors-on-vagrant-on-windows-because-the-paths-are-too-long/

**Comment by OmarShehata** [comment]
Ok, so applying the Vagrant symlink fix, and the Vagrant EPERM fix for filepaths that are too long, allows me to correctly `npm install superscript` and `npm install` inside my bot. 

But then I'm right back where I started with the OpenError: IO Error: sysmtemDB/MANIFEST-0000001.

**Comment by silentrob** [comment]
It sounds like you might be in a bad state, you can safely delete the `systemDB` folder. It is re-created on startup

**Comment by OmarShehata** [comment]
Thanks for the quick response! 

So I tried doing that, same error. I tried a completely fresh new VM. This time `bot-init` and `parse` work without having to change anything in the PATH. But I'm still getting the IOError with systemDB. 

Is there anything else that could be causing this? Or anything I can check?

**Comment by silentrob** [comment]
Try the steps in the `hello-superscript` repo. https://github.com/silentrob/hello-superscript. 

I'm assuming you are running a windows machine. I will also open an issue to make sure superscript works on windows.

**Comment by OmarShehata** [comment]
I'm running on a Windows 7 host machine but I'm trying to run superscript in Ubuntu 14 on Vagrant. I just try the `hello-superscript` repo, and when I run:

`./node_modules/superscript/bin/cleanup.js --mongo telnetbot` 

I get:

```
{ [Error: Cannot find module '../build/Release/bson'] code: 'MODULE_NOT_FOUND' }
users is clear
topics is clear
replies is clear
gambits is clear
events.js:72
   throw err; // Unhandled 'error' event 
OpenError: IO Error: ./systemDB/MANIFEST-000001: Invalid argument 
   at /vagrant/hello-superscript/node_modules/superscript/node_modules/sfacts/node_modules/level/node_modules/level-packager/node_modules/levelup/lib/levelup.js:114:34
```

Could this have anything to do with Mongo not running correctly or on the right port or anything? With Vagrant you have to specify which ports to bind/open. That's the only other thing I can think of that would be different from a pure Linux environment right now.

**Comment by silentrob** [comment]
What version of Node are you using? It should be 0.12.x or 5.11.x

**Comment by OmarShehata** [comment]
I was really hoping that would be it! I was using 0.10, I started from scratch again with node 0.12.14 and got everything working up to that point, but in `hello-superscript` I get the same error but without this part;

```
{ [Error: Cannot find module '../build/Release/bson'] code: 'MODULE_NOT_FOUND' }
js-bson: Failed to load c++ bson extension, using pure JS version
```

And the error is at events.js:85 now.

**Comment by silentrob** [comment]
Have a look at this thread. It seems to be related to mongo. http://stackoverflow.com/questions/28651028/cannot-find-module-build-release-bson-code-module-not-found-js-bson

**Comment by OmarShehata** [comment]
Sorry if my previous message wasn't clear. I meant updating node fixed the "Cannot find module bson" error. So I was only getting the OpenError IO Error SystemDB. 

The good news is I figured it out! The final problem was yet again a Vagrant issue where it can't access the file because of a filesystem issue. It's discussed in more detail here: https://github.com/Level/levelup/issues/222 

What worked for me is to just move my project out of the shared folder. So instead of having it in `/vagrant/mybot` I moved it to `/home/mybot`

Thanks for all your help @silentrob !
